htdocs/resetpassword.php

```diff
@@ -66,6 +97,40 @@
                 $posthook_message = $posthook_output[0];
             }
         }
+
+        #==============================================================================
+        # Notify password change
+        #==============================================================================
+        if ($result === "passwordchanged") {
+
+            if ($notify_on_change) {
+                # Search for user
+                $search = ldap_read($ldap, $dn, '(objectClass=*)', $mail_attributes);
+                $errno = ldap_errno($ldap);
+                if ( $errno ) {
+                    $result = "ldaperror";
+                    error_log("LDAP - Search error $errno  (".ldap_error($ldap).")");
+                } else {
+                    # Get user DN
+                    $entry = ldap_first_entry($ldap, $search);
+
+                    $mail = \Ltb\AttributeValue::ldap_get_mail_for_notification($ldap, $entry);
+                    if ($mail) {
+                        $data = array( "login" => $login, "mail" => $mail, "password" => $newpassword);
+                        if ( !\Ltb\Mail::send_mail_global($mail, $mail_from, $mail_from_name, $messages["changesubject"], $messages["changemessage"].$mail_signature, $data) ) {
```

Comment:
$login is empty here.

We should see how to merge the search attributes code with the one of the posthook just above, to avoid doing 2 LDAP searches if we have a posthook and a mail to send.

Reply:
I kept the 2 searches but added a parameter to get user name.